provider-blackduck/src/main/java/com/synopsys/integration/alert/provider/blackduck/processor/message/service/policy/BlackDuckComponentPolicyDetailsCreator.java

```diff
@@ -30,11 +32,12 @@ public BlackDuckComponentPolicyDetailsCreator(BlackDuckPolicySeverityConverter p
         this.policySeverityConverter = policySeverityConverter;
     }
 
-    public ComponentPolicy toComponentPolicy(ComponentPolicyRulesView componentPolicyRulesView) {
+    public ComponentPolicy toComponentPolicy(ComponentPolicyRulesView componentPolicyRulesView, Optional<PolicyRuleView> policyRuleView) {
         ComponentConcernSeverity componentConcernSeverity = policySeverityConverter.toComponentConcernSeverity(componentPolicyRulesView.getSeverity().name());
         boolean overridden = ProjectVersionComponentPolicyStatusType.IN_VIOLATION_OVERRIDDEN.equals(componentPolicyRulesView.getPolicyApprovalStatus());
         boolean vulnerabilityPolicy = isVulnerabilityPolicy(componentPolicyRulesView);
-        return new ComponentPolicy(componentPolicyRulesView.getName(), componentConcernSeverity, overridden, vulnerabilityPolicy, componentPolicyRulesView.getDescription());
+        String category = policyRuleView.map(ruleView -> ruleView.getCategory().name()).orElse(null);
```

Comment:
This can be mapped a second time instead of using a lambda.
```suggestion
        String category = policyRuleView.map(PolicyRulesView::getCategory).map(PolicyRuleCategoryType::name).orElse(null);
```